File: presentation/src/main/java/js/task/screens/main/list/adapter/ItemsListViewAdapter.java
package js.task.screens.main.list.adapter;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.util.List;
import java.util.function.Consumer;
import js.task.application.databinding.ListItemBinding;
import js.task.screens.main.list.holder.DataViewHolder;
import js.task.screens.main.list.model.PlaceholderItem;


public class ItemsListViewAdapter
        extends RecyclerView.Adapter<DataViewHolder>
{
    private final List<PlaceholderItem> data;
    private final Consumer<PlaceholderItem> callBack;

    public ItemsListViewAdapter(
            List<PlaceholderItem> data,
            Consumer<PlaceholderItem> callBack
    )
    {
        this.data = data;
        this.callBack = callBack;
    }

    @NonNull
    @Override
    public DataViewHolder onCreateViewHolder(
            @NonNull ViewGroup parent,
            int viewType
    )
    {
        ListItemBinding binding = ListItemBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
        return new DataViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(
            @NonNull DataViewHolder holder,
            int position
    )
    {
        holder.onBindData(data.get(position), callBack);
    }

    @Override
    public int getItemCount()
    {
        return data.size();
    }
}

File: presentation/src/main/java/js/task/screens/main/list/adapter/ItemsListViewAdapter.kt
package js.task.screens.main.list.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import js.task.application.databinding.ListItemBinding
import js.task.screens.main.list.holder.DataViewHolder
import js.task.screens.main.list.model.PlaceholderItem


open class ItemsListViewAdapter(
    private val data : List<PlaceholderItem>, private val callBack : (PlaceholderItem) -> Unit
) : RecyclerView.Adapter<DataViewHolder>()
{
    override fun onCreateViewHolder(parent : ViewGroup, viewType : Int) : DataViewHolder
    {
        val binding = ListItemBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return DataViewHolder(binding)
    }

    override fun getItemCount() = data.size

    override fun onBindViewHolder(holder : DataViewHolder, position : Int)
    {
        holder.onBindData(data[position], callBack)
    }
}